traitlets/traitlets.py

```diff
@@ -644,6 +648,17 @@ def hold(*a):
                 for v in cache.values():
                     self._notify_trait(*v)
 
+    @contextlib.contextmanager
+    def _allow_write(self):
```

Comment:
- In widgets, the context manager is used to allow the widget machinery to write when receiving state from the js side. It is only read-only for the Python user.
- In the usecase with sensors, the thread polling the sensors will use that context manager to write sensor values.